### PR TITLE
Add retryable

### PR DESCRIPTION
**Pull request by ybiquitous**
- https://rubygems.org/gems/retryable
- https://github.com/nfedyashev/retryable

~~NOTE: This change updates `Gemfile` and `Gemfile.lock` because the branches `repo` and `update-rbs` do not exist already in `ruby/rbs` and `soutaro/steep`.~~
=> Merged the latest `Gemfile*` files from the `main` branch.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source "https://rubygems.org"
 
-gem "rbs", git: "https://github.com/ruby/rbs.git", branch: "repo"
```
Comment (soutaro) [inline Gemfile:3]: 🙇‍♂️

**Comment by soutaro** [comment]
@ybiquitous I recommend adding some tests from examples in [README.md of the gem](https://github.com/nfedyashev/retryable/blob/master/README.md).

File: gems/retryable/3.0/_test/test.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require "retryable"
+
+Retryable.retryable do
+  puts "foo"
+end
+
+Retryable.retryable(tries: 3, on: ArgumentError) do
```
Comment (ybiquitous) [inline gems/retryable/3.0/_test/test.rb:7]: Oh, sorry for my laziness... 😓 
When I've added a test case, but the following error occurs:

```console
$ bundle exec steep check
test.rb:7:20: ArgumentTypeMismatch: receiver=singleton(::Retryable), expected=::Retryable::Configuration::options, actual=::Hash[::Symbol, (::Integer | singleton(::ArgumentError))] (tries: 3, on: ArgumentError)
```

Can you provide any solutions to fix the error, please? The following locations are related to the error:

- https://github.com/soutaro/gem_rbs/pull/1/files#diff-e6bd80a44522ce1c893565fac2966e94b970b8001786e19f5a8f993cb4213120R38
- https://github.com/soutaro/gem_rbs/pull/1/files#diff-f556dfb0a545334e2fba3e9b5b1b3321129c2a93cd8a55b935b3d917827938adR15

Comment (soutaro) [reply gems/retryable/3.0/_test/test.rb:7]: Steep 0.17.0 would help. https://github.com/soutaro/steep/pull/256

It allows omitting `nil`-able attributes, so defining options as the following can be an option.

```
    type options = {
      contexts: Hash[Symbol, options]?,
      ensure: ^(Integer) -> void | nil,
      exception_cb: ^(Exception) -> void | nil,
      log_method: ^(Integer, Exception) -> void | nil,
      matching: (String | Regexp) | Array[String | Regexp] | nil,
      not: _Exception | Array[_Exception] | nil,
      on: _Exception | Array[_Exception] | nil,
      sleep: real | (^(real) -> void) | nil,
      sleep_method: ^(real) -> void | nil,
      tries: Integer | nil
    }
```

Comment (soutaro) [reply gems/retryable/3.0/_test/test.rb:7]: つ https://github.com/soutaro/gem_rbs/pull/2 🙇

Comment (ybiquitous) [reply gems/retryable/3.0/_test/test.rb]: Thank you! I've updated `Gemfile*` files to the latest version and fixed the test!

**Review by soutaro — APPROVED** [review]
🙌

**Comment by ybiquitous** [comment]
@soutaro Thank you so much for your help! 😊

**Pull request merged**